src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -51,6 +51,18 @@
 
     /**
      * Constructs an unauthenticated BoxAPIRequest.
+     *
+     * @param url
+     *            the {@link URL} of the request
+     * @param method
+     *            the {@link HttpMethod} of the request
+     */
+    public BoxAPIRequest(URL url, HttpMethod method) {
```

Comment:
Clean code suggests we should avoid String types and use Enum in such cases to follow fail early principle (during the compilation). I would lean to deprecation of the constructor with String. On the other hand, these kind of "issues" could be handled during the 'Box Java SDK feedback and improvement process' :)